### PR TITLE
Add a date to the header document

### PR DESCRIPTION
**Pull request by iainlane**
& avoid updating the output more than is necessary so that the file is stable.

This is so we can track if the remote side (dak or launchpad and apt) is pulling from us correctly.

### CODE REVIEW

File: dep11/generator.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def extract_metadata(mde, sn, pkg):
     cpts = mde.process(pkg)
 
     msgtxt = "Processed: %s (%s/%s), found %i" % (pkg.name, sn, pkg.arch, len(cpts))
-    return msgtxt
+    return (msgtxt, len(cpts) > 0)
```
Comment (iainlane) [inline dep11/generator.py]: Is this check enough or could we do more?

File: dep11/component.py [hunk 2]
```diff
@@ -30,10 +30,11 @@
 
 ###########################################################################
 DEP11_VERSION = "0.8"
-time_str = str(datetime.date.today())
+now = datetime.datetime.now(datetime.timezone.utc).isoformat()
```
Comment (iainlane) [inline dep11/component.py:33]: FYI it's this and not utcnow() because this way you get the timezone set in the output.

**Comment by iainlane** [comment]
OK I just pushed some updates, please re-review. I wasn't sure that I understood the comment about the hints file correctly (don't know why this is necessary), so that's a separate commit - you can squash them when merging if you want. The field is now called `Time` and we use `has_ignore_reason` to see if there is new stuff.

**Comment by ximion** [comment]
Looks good to me! Sorry for not being able to merge this today, will do it tomorrow!
About the hints: Well, if we had exactly zero new components, we don't need to re-write them either, but that would need a second check. One for "are there any valid components?" and one for "are there any invalid components?".

**Comment by iainlane** [comment]
Ah right, because hints describes the errors - I get it. And there's already a check to skip the suite in the "0 new components" case, so I think we are good.

**Comment by iainlane** [comment]
(conflict resolved)

**Comment by ximion** [comment]
Patch applied to master,
Thanks!

**Pull request closed without merging**